lib/samples/validate_utility_network_topology/validate_utility_network_topology.dart

```diff
@@ -0,0 +1,700 @@
+                    style: Theme.of(context).textTheme.labelLarge,
+                  ),
+                ],
+              ),
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+      // Configure the bottom sheet the display an attribute picker when required.
+      bottomSheet: _attributePickerVisible ? buildAttributePicker() : null,
+    );
+  }
+
+  Future<void> onMapViewReady() async {
+    // Create a map using the portal item of a web map containing a utility network.
+    final portal = Portal(
+      Uri.parse('https://sampleserver7.arcgisonline.com/portal/sharing/rest'),
+    );
+    final portalItem = PortalItem.withPortalAndItemId(
+      portal: portal,
+      itemId: '6e3fc6db3d0b4e6589eb4097eb3e5b9b',
+    );
+    _map = ArcGISMap.withItem(portalItem);
+
+    // Set an initial viewpoint on the map.
+    _map.initialViewpoint = Viewpoint.fromTargetExtent(
+      Envelope.fromXY(
+        xMin: -9815489.0660101417,
+        yMin: 5128463.4221229386,
+        xMax: -9814625.2768726498,
+        yMax: 5128968.4911854975,
+        spatialReference: SpatialReference.webMercator,
+      ),
+    );
+
+    // Load in persistent session mode (workaround for server caching issue):
+    // https://support.esri.com/en-us/bug/asynchronous-validate-request-for-utility-network-servi-bug-000160443
+    _map.loadSettings = LoadSettings()
+      ..featureServiceSessionType = FeatureServiceSessionType.persistent;
+
+    // Load the map.
+    await _map.load();
+
+    // Define labels on the map for visualizing attribute editing.
+    defineLabelsForLayer(_deviceTableName, _deviceStatusField, Colors.indigo);
+    defineLabelsForLayer(_lineTableName, _nominalVoltageField, Colors.red);
+
+    // Configure the utility network and trace parameters.
+    try {
+      await configureUtilityNetwork();
+    } on ArcGISException catch (e) {
+      setState(() {
+        _ready = true;
+        _statusTitle = 'Failed to configure utility network.';
+        _statusDetail = e.message;
+      });
+      return;
+    }
+
+    // Add the map to the map view.
+    _mapViewController.arcGISMap = _map;
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() {
+      _ready = true;
+      _statusTitle = 'Utility Network loaded';
+      _statusDetail =
+          "Tap on a feature to edit.\nTap 'Get State' to check if validating is necessary or if tracing is available.\nTap 'Trace' to run a trace.";
+    });
+  }
+
+  Future<void> configureUtilityNetwork() async {
+    setState(() => _statusTitle = 'Loading utility network...');
+
+    // Get the utility network from the map and load.
+    _utilityNetwork = _map.utilityNetworks.first;
+    await _utilityNetwork.load();
+
+    // Get the service geodatabase.
+    final serviceGeodatabase = _utilityNetwork.serviceGeodatabase;
```

Comment:
I would just use `!` at this point.
```dart
    final serviceGeodatabase = _utilityNetwork.serviceGeodatabase!;
```
We haven't really discussed it, but my approach would be: this client was written with a specific webmap in mind, and we control both the client and the webmap. Therefore if the webmap isn't authored as designed, it is actually correct for the client to fail spectacularly, such as by using `!` on a null value. So wherever this client app has a hard prerequisite that some object must be available, we should use `!`. That would cover `serviceGeodatabase` here, as well as `terminalConfiguration`, `domainNetwork`, and `sourceTier` below. We should `!` those as early as possible, wherever we are relying on the webmap to have been authored with them.